ale/base/data_naif.py

```diff
@@ -322,23 +322,77 @@ def sensor_position(self):
+                state, _ = spice.spkezr(self.spacecraft_name,
                                        time,
                                        self.reference_frame,
                                        self.light_time_correction,
-                                       self.spacecraft_name,)
+                                       self.target_name,)
                 pos.append(state[:3])
                 vel.append(state[3:])
             # By default, spice works in km, and the vector returned by spkezr points the opposite
             # direction to what ALE needs, so it must be multiplied by (-1)
-            self._position = [p * -1000 for p in pos]
-            self._velocity = [v * -1000 for v in vel]
+            self._position = [p * 1000 for p in pos]
+            self._velocity = [v * 1000 for v in vel]
         return self._position, self._velocity, self.ephemeris_time
 
+    @property
+    def frame_trace(self):
+        frame_codes = [self.sensor_frame_id]
+        _, frame_type, _ = spice.frinfo(frame_codes[-1])
+        frame_types = [frame_type]
+
+        while(frame_codes[-1] != 1):
+            try:
+                center, frame_type, frame_type_id = spice.frinfo(frame_codes[-1])
+            except Exception as e:
+                print(e)
+                break
+
+            if frame_type is 1 or frame_type is 2:
+                frame_code = 1
+
+            elif frame_type is 3:
+                matrix, frame_code, found = spice.ckfrot(frame_type_id, self.center_ephemeris_time)
+                if not found:
+                    raise Exception(f"The ck rotation from frame {frame_codes[-1]} can not \
+                                      be found due to no pointing available at requested time \
+                                      or a problem with the frame")
+            elif frame_type is 4:
+                matrix, frame_code, found = spice.tkfram(frame_type_id)
+                if not found:
+                    raise Exception(f"The tk rotation from frame {frame_codes[-1]} can not \
+                                      be found")
+            elif frame_type is 5:
+                matrix, frame_code = spice.zzdynrot(frame_type_id, center, self.center_ephemeris_time)
+
+            else:
+                raise Exception(f"The frame {frame_codes[-1]} has a type {frame_type_id} \
+                                  not supported by your version of Naif Spicelib. \
+                                  You need to update.")
+
+            frame_codes.append(frame_code)
+            frame_types.append(frame_type)
+        constant_frames = [frame_codes[i] for i, frame_type in enumerate(frame_types) if frame_type == 4]
+
+        time_dependent_frames = []
+        if len(constant_frames) != 0:
+            time_dependent_frames.append(constant_frames[-1])
+
+        for i, frame_type in enumerate(frame_types):
+            if frame_type != 4:
+                time_dependent_frames.append(frame_codes[i])
```

Comment:
This will result in losing the order of the frame chain. For example, if the sensor frame chain is J2000 --ck--> target --ck--> spacecraft --fixed--> platform --ck--> gimble --fixed--> sensor then you will get:

constant_frames = [sensor, platform]
time_dependent_frames = [platform, gimble, spacecraft, target, J2000]

The way ISIS defines these is the time_dependent_frames is the rotation from J2000 to the last time dependent frame and then constant_frames is the rest. So, in ISIS this would look like:

constant_frames = [sensor, gimble]
time_dependent_frames = [gimble, platform, spacecraft, target, J2000]

Reply:
:+1: Will get this fixed up